includes/sanitizers/class-amp-iframe-sanitizer.php

```diff
@@ -135,8 +135,20 @@ public function sanitize() {
 				if ( $node->parentNode->parentNode instanceof DOMElement && 'figure' === $node->parentNode->parentNode->tagName ) {
 					$figure_node = $node->parentNode->parentNode;
 				}
-				if ( $figure_node && $figure_node->hasAttribute( 'class' ) && in_array( 'alignfull', explode( ' ', $figure_node->getAttribute( 'class' ) ), true ) ) {
-					$normalized_attributes['layout'] = 'responsive';
+
+				if ( $figure_node && $figure_node->hasAttribute( 'class' ) ) {
+
+					$figure_node_classes = explode( ' ', $figure_node->getAttribute( 'class' ) );
+
+					// If the alignment was set to 'wide width' or 'full width', set the layout to responsive.
+					$show_responsive = (
+						in_array( 'alignfull', $figure_node_classes, true ) ||
+						in_array( 'alignwide', $figure_node_classes, true )
+					);
+
+					if ( true === $show_responsive ) {
```

Comment:
I think this can be simplified a bit:

```suggestion
					if ( in_array( 'alignfull', $figure_node_classes, true ) || in_array( 'alignwide', $figure_node_classes, true ) ) {
```